src/main/java/uk/gov/companieshouse/officerfiling/api/validation/AddressValidator.java

```diff
@@ -0,0 +1,171 @@
+        super(logger, companyProfileService, inputAllowedNationalities, apiEnumerations);
+        this.countryList = countryList;
+        this.ukCountryList = ukCountryList;
+    }
+
+    @Override
+    public void validateRequiredDtoFields(HttpServletRequest request, List<ApiError> errorList, OfficerFilingDto dto) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    public void validateOptionalDtoFields(HttpServletRequest request, List<ApiError> errorList, OfficerFilingDto dto) {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * Validate the given address, append any validation errors to the given errorList.
+     *
+     * @param addressErrorProvider Specifies which error messages will be used if any validation errors occur
+     * @param request              Used to attach request information to any errors raised
+     * @param errorList            All validation errors triggered will be appended to this list
+     * @param address              The address being validated
+     */
+    public void validate(AddressErrorProvider addressErrorProvider, HttpServletRequest request, List<ApiError> errorList, AddressDto address) {
+        if (address == null) {
+            createBlankValidationErrors(addressErrorProvider, request, errorList);
+            return;
+        }
+        validatePremises(addressErrorProvider, request, errorList, address.getPremises());
+        validateAddressLineOne(addressErrorProvider, request, errorList, address.getAddressLine1());
+        validateAddressLineTwo(addressErrorProvider, request, errorList, address.getAddressLine2());
+        validateLocality(addressErrorProvider, request, errorList, address.getLocality());
+        validateRegion(addressErrorProvider, request, errorList, address.getRegion());
+        validateCountry(addressErrorProvider, request, errorList, address.getCountry());
+        validatePostalCode(addressErrorProvider, request, errorList, address.getPostalCode(), address.getCountry());
+    }
+
+    private void createBlankValidationErrors(AddressErrorProvider addressErrorProvider, HttpServletRequest request, List<ApiError> errorList) {
+        createValidationError(request, errorList, addressErrorProvider.getPremisesBlank());
+        createValidationError(request, errorList, addressErrorProvider.getAddressLineOneBlank());
+        createValidationError(request, errorList, addressErrorProvider.getLocalityBlank());
+        createValidationError(request, errorList, addressErrorProvider.getPostalCodeBlank());
+        createValidationError(request, errorList, addressErrorProvider.getCountryBlank());
+    }
+
+    private void validatePremises(AddressErrorProvider addressErrorProvider, HttpServletRequest request, List<ApiError> errorList, String premises) {
+        if (premises == null || premises.isBlank()) {
```

Comment:
You could use if(StringUtils.isBlank(premises)) 
that avoid the null check and cuts down testing.